sync.py

```diff
@@ -32,348 +26,187 @@
-
-                details = cohort_details[j]['structuredDataNodes']['structuredDataNode']
-
-                delivery_code = row['delivery_code']
-                delivery_label = row['delivery_label']
-                if not delivery_label:
-                    try:
-                        delivery_label = labels[delivery_code]
-                    except KeyError:
-                        delivery_label = ""
-                try:
-                    delivery_row_code = delivery_descriptions[row['delivery_code']]
-                except KeyError:
-                    delivery_row_code = ""
-
-                self.find(details, 'delivery_description')['text'] = delivery_row_code
-                self.find(details, 'delivery_label')['text'] = delivery_label
-
-                # adding delivery sub headings
-                try:
-                    delivery_subheadings = subheadings[delivery_code]
-                except KeyError:
-                    delivery_subheadings = ""
-                if self.find(details, 'delivery_subheading'):
-                    self.find(details, 'delivery_subheading')['text'] = delivery_subheadings
-                else:
-                    details.append({'text': delivery_subheadings, 'identifier': 'delivery_subheading', 'type': 'text'})
-
-                try:
-                    location = locations[row['location']]
-                except KeyError:
-                    location = ""
-                    self.missing_locations.append(row['location'])
-
-                if delivery_code in ['O', 'OO']:
-                    location = 'Online'
-
-                self.find(details, 'location')['text'] = location
-
-                if ';' in row['start_date']:
-                    # turn '4/17/2017;  05/29/2017; 07/10/2017' into usefulness
-                    calendar = datetime.datetime.strptime(row['start_date'], "%m/%d/%Y;")
-                    calendar = calendar.strftime("%m-%d-%Y")
-                    self.find(details, 'cohort_start_type')['text'] = "Calendar"
-                    self.find(details, 'calendar_start')['text'] = calendar
-                    self.find(details, 'semester_start')['text'] = ""
-                    self.find(details, 'year_start')['text'] = ""
+                    new_cohort = copy.deepcopy(old_cohort)
+                    find(concentration, 'concentration_banner', False).append(new_cohort)
+
+                # set the new cohort details - if we get a list from bu_cascade, get the first element?
+                new_cohort_details = find(concentration, 'cohort_details')
+                if isinstance(new_cohort_details, list):
+                    new_cohort_details = new_cohort_details[-1]
+
+                # clear values
+                for to_clear in new_cohort_details['structuredDataNodes']['structuredDataNode']:
+                    to_clear['text'] = ''
+
+                if row['start_term_code'] == u'000000':
```

Comment:
This if statement might be in need of a comment; why do we do these things when the start code is 000000?

Reply:
The comment is the next line. I personally don't prefer to have comments before an else. But I would be fine moving it if you think that would help

Reply:
Is `# start dates (or dynamic)` the comment you're referring to? I saw that comment, but didn't put together that code 000000 indicated that the block was supposed to use dynamic dates. I thought that comment was about the two `update()` calls below it.

Reply:
no worries. i'll move around the comment